Implement agent solvers `claude_code()` and `codex_cli()` (#218)
simonpcouch/rinspect: vignettes/articles/solvers.Rmd

If we had evaluated this model across multiple epochs, the question ID could become a "nuisance parameter" in a mixed model, e.g. with the model structure `ordinal::clmm(score ~ model + (1|id), ...)`.
:::

## Evaluating coding agents

The solvers above are harnesses you assemble yourself with ellmer. To see how an off-the-shelf coding agent stacks up on the same problems, vitals provides the solvers `claude_code()` and `codex()`, which evaluate the Claude Code and Codex agents on a task's dataset:

```{r}
#| label: are-claude-code
#| eval: false
are_claude_code <- Task$new(
  dataset = are,
  solver = claude_code(model = "anthropic/claude-sonnet-4-5"),
  scorer = model_graded_qa(partial_credit = TRUE, scorer_chat = sonnet_3_7$clone()),
  name = "An R Eval"
)

are_claude_code$eval()
```

> These solvers run the agent's command line interface in a Docker sandbox via Python Inspect (resolved automatically with reticulate—you'll need the reticulate package installed and a running Docker daemon). The agent's model calls are proxied, so your API key never enters the sandbox, and its transcript is read back into ellmer Chat objects: tool calls appear in the log viewer, and token usage is tallied as usual.

Author: ```suggestion
These solvers run the agent's command line interface in a Docker sandbox via Python Inspect (resolved automatically with reticulate—you'll need the reticulate package installed and a running Docker daemon). The agent's transcript is read back into ellmer Chat objects and logged as usual.
```

If we had evaluated this model across multiple epochs, the question ID could become a "nuisance parameter" in a mixed model, e.g. with the model structure `ordinal::clmm(score ~ model + (1|id), ...)`.
:::

## Evaluating coding agents

The solvers above are harnesses you assemble yourself with ellmer. To see how an off-the-shelf coding agent stacks up on the same problems, vitals provides the solvers `claude_code()` and `codex()`, which evaluate the Claude Code and Codex agents on a task's dataset:

```{r}
#| label: are-claude-code
#| eval: false
are_claude_code <- Task$new(
  dataset = are,
  solver = claude_code(chat_claude(model = "claude-sonnet-4-5")),
  scorer = model_graded_qa(partial_credit = TRUE, scorer_chat = sonnet_3_7$clone()),
  name = "An R Eval"
)

are_claude_code$eval()
```

These solvers run the agent's command line interface in a Docker sandbox via Python Inspect (resolved automatically with reticulate—you'll need the reticulate package installed and a running Docker daemon). The agent's transcript is read back into ellmer Chat objects and logged as usual.

> Each sample gets its own throwaway container, built from Inspect's default image unless you supply a `Dockerfile` or `compose.yaml`. For a dataset like `are`, whose questions are self-contained, that empty workspace is all the agent needs; to evaluate an agent on a repository, build it into the image and point the agent at it with `cwd`.

Author: Could we write out what `cwd` is an argument to? e.g. `claude_code(cwd)`?
